omegaconf/grammar/OmegaConfGrammarParser.g4

```diff
@@ -41,13 +41,14 @@ sequence: (element (COMMA element?)*) | (COMMA element?)+;
 // Interpolations.
 
 interpolation: interpolationNode | interpolationResolver;
+
 interpolationNode:
       INTER_OPEN
       DOT*                                                       // relative interpolation?
       (
         (configKey | BRACKET_OPEN configKey BRACKET_CLOSE)       // foo, [foo]
         (DOT configKey | BRACKET_OPEN configKey BRACKET_CLOSE)*  // .foo, [foo], .foo[bar], [foo].bar[baz]
-      )?
+      )
```

Comment:
Probably the parentheses can be removed?
Also can you update grammar_visitor:
`assert ctx.getChildCount() >= 2` should become `assert ctx.getChildCount() >= 3` and the comment above that assert should be updated with the new definition

Reply:
+ the regex version and associated tests probably need a look too

Reply:
Yeah. I took a look at the regex pattern and it wasn't obvious to me how to fix it (this is no longer a simple revert because of the addition of brackets to the node path syntax). can you take a look? (I moved the relevant passing cases to the negative test).

Reply:
I pushed it directly, hope that's ok:
* cb902788d25f7ac5ed1dc4d2d7bbe818800e7083 adds a consistency check between the regex and the full grammar
* 9fe900e9eb7b75d73f51ef5be368d117359b91e6 is the actual fix to the regex

Reply:
yeah, it's fine - thanks.